### PR TITLE
updated publish_object to reformat sysmeta fileNames

### PR DESCRIPTION
**Pull request by dmullen17**
Updated publish_object to follow the new naming conventions outlined here: https://github.com/NCEAS/metacatui/issues/915. 

Close#126

### CODE REVIEW

**Comment by dmullen17** [comment]
@jeanetteclark thanks for looking over this!  The function removes special characters from the system metadata fileName slot and returns a reformatted version.  It stores the extension based on the system metadata `formatId`, replaces special characters with underscores, trims the end of the string to remove the reformatted extension, and finally pastes the original back on.

File: R/editing.R [hunk 1]
```diff
@@ -979,3 +979,36 @@ update_package_object <- function(mn,
...
+#' @param sysmeta (S4) A system metadata object
+#'
+reformat_file_name <- function(path, sysmeta) {
+  base_name <- basename(path)
+  if (sysmeta@formatId == 'http://www.openarchives.org/ore/terms') {
+    ext <- '.rdf.xml'
+  } else if (grepl('eml://ecoinformatics\\.org/eml*', sysmeta@formatId)) {
+    ext <- '.xml'
+    # remove extension then truncate to 50 characters
+    base_name <- stringr::str_sub(base_name, end = -5) %>%
```
Comment (jeanetteclark) [inline R/editing.R]: you could use `tools::file_path_sans_ext` to more predictably remove just the extension here as opposed to the last 5 characters of the filename (though it really doesn't make a difference functionally)

Comment (dmullen17) [reply R/editing.R]: I like that change - definitely more elegant

**Comment by dmullen17** [comment]
Close #126

**Pull request merged**